### PR TITLE
youtube-cleanup: add option to disable ambient mode

### PR DESCRIPTION
**Pull request by Dosenpfand**
Ambient mode in YouTube can be annoying and resource consuming.  This PR adds an option to disable it in the YouTube Cleanup filter template.
More information about ambient mode: https://www.ghacks.net/2022/10/25/how-to-disable-youtubes-new-ambient-mode-playback-feature/

### CODE REVIEW

**Comment by JohnyP36** [comment]
Is this ambient mode only for the mobile version of YouTube or also for the desktop version.

**Comment by xvello** [comment]
Hello @Dosenpfand, thanks for your contribution!

My main gripe with ambient mode is the unreasonable CPU usage that it creates. I'll need to properly benchmark that, but I'm afraid that this rule will hide the resulting gradient, without killing the javascript that updates it. This is the reason why when hiding the chat, [we :remove() it](https://github.com/letsblockit/letsblockit/blob/61719158b423b3de6cb84db161e8da0640c25c14/data/filters/templates/youtube-cleanup.yaml#L58-L60), to get the javascript to fail and stop pulling messages. Maybe the same trick could work here, we'll have to benchmark it.

If it does not do the trick and the CPU cost is still here, I don't think we're doing a service to our users by adding this option, instead of directing them to the toggle to disable it in the player UI.

--- 
@JohnyP36 ambient mode is also available on desktop, but not if the player is in the wide "theater mode" 
image

File: data/filters/templates/youtube-cleanup.yaml [hunk 1]
```diff
@@ -66,6 +70,9 @@ template: |
   {{! Applied to both www. and m. sites, defuses custom pan events for swipe }}
   youtube.com##+js(aeld, /^pan(start|move|end)$/)
   {{/if}}
+  {{#if no-ambient-mode}}
+  www.youtube.com###cinematics.ytd-watch-flexy
```
Comment (xvello) [inline data/filters/templates/youtube-cleanup.yaml:74]: From my benchmarking with Firefox on macOS, your rule does only negate half of the CPU overhead of the ambient mode: the gradient does not need to be displayed, but it is still updated by javascript code.

Using `:remove()` seems like it completely nullifies the CPU cost, probably by making the javascript loop fail to start. Interestingly, it even removes the parameter item.

```suggestion
  www.youtube.com###cinematics.ytd-watch-flexy:remove()
```

**Review by xvello — CHANGES_REQUESTED** [review]

**Review by xvello — DISMISSED** [review]
Heya @Dosenpfand, thanks for your contribution! I took the liberty to patch the rule as discussed, and am releasing it right now.

Ideally, we'd also write a rule for the mobile web interface, and add it to the same option, but we can do this in another PR.

**Comment by Dosenpfand** [comment]
Hi @xvello thanks for your quick and valuable feedback! Next time I will take into account benchmarking/profiling myself.

**Review by xvello — APPROVED** [review]

**Comment by xvello** [comment]
Heya @Dosenpfand, the new option is available at https://letsblock.it/filters/youtube-cleanup, thanks again for your contribution!

**Pull request merged**